code/controllers/ModelAsController.php

```diff
@@ -49,82 +49,79 @@ public function init() {
 	public function handleRequest(SS_HTTPRequest $request, DataModel $model) {
 		$this->setRequest($request);
 		$this->setDataModel($model);
-		
-		$this->pushCurrent();
 
-		// Create a response just in case init() decides to redirect
-		$this->response = new SS_HTTPResponse();
```

Comment:
This is a regression in the framework PR too...

Storing $this->getResponse() as a local var doesn't solve any problem directly. it's the calling of `getResponse()` that will fix the regression